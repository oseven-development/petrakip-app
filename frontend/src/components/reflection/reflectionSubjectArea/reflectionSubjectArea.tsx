import React from 'react'

import {
  IonList,
  IonText,
  IonItem,
  IonLabel,
  IonBadge,
  IonIcon,
  IonButton,
} from '@ionic/react'

import {
  badgeStatus,
  checkmarkIcon,
  currentSelectedTopic,
  doneStatusText,
  textLable,
} from './reflectionSubjectAreaUtils'

interface TopicItem {
  id: string
  lable: string
  describe: string
  subListItems: SubTopicItem[]
}

export interface SubTopicItem {
  id: string
  lable: string
  done: boolean
}

export type InputSelectValue = string | 'Später Wählen'

interface Props {
  selectValue: (input: InputSelectValue) => void
  topicList: TopicItem[]
}

/*
    The Component is designed to not store the selected value in the second level
    You must Provide a Callback-Function where the value is emitted to and store
    the state outside of this Component
 */
export const SubjectArea: React.FC<Props> = ({ topicList, selectValue }) => {
  // this state stores the current selected row
  // it is the id or undefined if nothing is selected
  const [currentSelected, setCurrentSelected] = React.useState<
    undefined | string
  >(undefined)

  const selectTopic = React.useCallback(
    (
      event:
        | React.MouseEvent<HTMLIonItemElement>
        | React.MouseEvent<HTMLIonButtonElement, MouseEvent>,
    ) => {
      //@ts-ignore
      const selectedTopic: string = event.target.getAttribute('data-id')

      setTimeout(() => {
        selectValue(selectedTopic)
      }, 300)
    },
    [selectValue],
  )

  // Set the currentSelected value to the clicked id
  // if the id is Already selected, set the current selected value to undefined
  // this Behavior is needed to unselect the row
  const handlerCurrentSelected = (
    event: React.MouseEvent<HTMLIonItemElement>,
  ): void => {
    //@ts-ignore
    const clickedRow = event.target.getAttribute('data-id')
    setCurrentSelected(currentSelected =>
      currentSelected === clickedRow ? undefined : clickedRow,
    )
  }

  const SubSubjectAreaList: React.FC<{ subListItems: SubTopicItem[] }> = ({
    subListItems,
  }) => {
    return (
      <IonList>
        {subListItems.map(({ lable, done, id }, subIndex) => (
          <IonItem
            button
            data-test={id}
            data-id={lable}
            onClick={selectTopic}
            key={`reflection-topic-list-${lable}-${subIndex}`}
          >
            <IonLabel data-id={lable}>{lable}</IonLabel>
            <IonIcon
              data-id={lable}
              icon={checkmarkIcon(done).icon}
              color={checkmarkIcon(done).color}
              slot="end"
            />
          </IonItem>
        ))}
      </IonList>
    )
  }

  return (
    <>
      <IonList>
        {topicList.map(({ lable, id, describe, subListItems }, index) => (
          <React.Fragment key={`reflection-topic-list-${index}`}>
            <IonItem
              button
              data-test={id}
              data-id={id}
              onClick={handlerCurrentSelected}
              color={currentSelectedTopic(currentSelected, id)}
            >
              <IonLabel data-id={id} color={textLable(currentSelected, id)}>
                {lable}
              </IonLabel>
              <IonBadge data-id={id} color={badgeStatus(subListItems)}>
                {doneStatusText(subListItems)}
              </IonBadge>
            </IonItem>
            {/* render the sublist only when the current selected value are equally the id in the current loop */}
            {currentSelected === id && (
              <>
                <IonItem>
                  <IonText
                    color="medium"
                    className="ion-padding-top ion-padding-bottom"
                  >
                    {describe}
                  </IonText>
                </IonItem>
                <SubSubjectAreaList {...{ subListItems }} />
              </>
            )}
          </React.Fragment>
        ))}
      </IonList>
      <IonButton
        data-id={'Später wählen'}
        data-test={'choseLater'}
        onClick={selectTopic}
      >
        Später wählen
      </IonButton>
    </>
  )
}
