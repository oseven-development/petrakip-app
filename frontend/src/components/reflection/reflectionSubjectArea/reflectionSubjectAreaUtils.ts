import { checkmarkCircle, checkmarkCircleOutline } from 'ionicons/icons'

import { SubTopicItem } from './reflectionSubjectArea'

export function totalDone(subListItems: SubTopicItem[]): number {
  return subListItems.filter(
    ({ subjectStatusCompleted }) => subjectStatusCompleted,
  ).length
}

export function badgeStatus(
  subListItems: SubTopicItem[],
): 'medium' | 'primary' | 'success' {
  const done = totalDone(subListItems)
  const total = subListItems.length
  if (done === 0) {
    return 'medium'
  } else if (done !== total) {
    return 'primary'
  } else {
    return 'success'
  }
}

export function currentSelectedTopic(
  currentSelected: string | undefined,
  id: string,
): 'tertiary' | undefined {
  return currentSelected === id ? 'tertiary' : undefined
}

export function textLable(
  currentSelected: string | undefined,
  id: string,
): 'medium' | undefined {
  return currentSelected === undefined || currentSelected === id
    ? undefined
    : 'medium'
}

export function doneStatusText(subListItems: SubTopicItem[]) {
  return `${totalDone(subListItems)}/${subListItems.length}`
}

export function checkmarkIcon(
  done: boolean,
): { icon: string; color: 'success' | 'medium' } {
  const icon = done ? checkmarkCircle : checkmarkCircleOutline
  const color = done ? 'success' : 'medium'
  return { icon, color }
}
