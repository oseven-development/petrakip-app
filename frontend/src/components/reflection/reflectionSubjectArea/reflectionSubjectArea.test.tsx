import { render, act, fireEvent } from '@testing-library/react'

import { SubjectArea } from './reflectionSubjectArea'

import testDataSet from '../../../../test/reflectionsetClean'

describe('main suite', () => {
  beforeEach(() => {
    jest.useFakeTimers()
  })
  const selectTopic = (
    <SubjectArea topicList={testDataSet} selectValue={() => {}} />
  )

  test('renders without crashing', () => {
    const { container } = render(selectTopic)
    expect(container).toMatchSnapshot()
  })

  test('render sublist when click on 1 Fachinhalte', () => {
    const { container } = render(selectTopic)
    const el = container.querySelector('[data-test="1-fachinhalte"]')
    act(() => {
      if (el) fireEvent.click(el)
    })
    expect(container).toMatchSnapshot()
  })

  test('render sublist when click on 4 Unterrichtsstrategien', () => {
    const { container } = render(selectTopic)
    const el = container.querySelector('[data-test="4-unterrichtsstrategien"]')
    act(() => {
      if (el) fireEvent.click(el)
    })
    expect(container).toMatchSnapshot()
  })

  test('when click on chose later the function should emit chose later', () => {
    const mockFn = jest.fn()

    const { container } = render(
      <SubjectArea topicList={testDataSet} selectValue={mockFn} />,
    )

    const el = container.querySelector('[data-test="choseLater"]')
    act(() => {
      if (el) fireEvent.click(el)
    })

    jest.runAllTimers()
    expect(mockFn).toHaveBeenCalledTimes(1)
    expect(mockFn).toHaveBeenLastCalledWith('Später wählen')
  })

  test.each`
    mainElementId                  | subElementId                    | result
    ${'5-motivieren-und-anleiten'} | ${'5.3-zeitmanagement'}         | ${'5.3 Zeitmanagement'}
    ${'7-planen-und-evaluieren'}   | ${'7.1-leitideen-und-lehrplan'} | ${'7.1 Leitideen und Lehrplan'}
  `(
    '`when click on $subElementId the function should emit $result`',
    ({ mainElementId, subElementId, result }) => {
      const mockFn = jest.fn()

      const { container } = render(
        <SubjectArea topicList={testDataSet} selectValue={mockFn} />,
      )
      const el = container.querySelector(`[data-test="${mainElementId}"]`)
      act(() => {
        if (el) fireEvent.click(el)
      })

      const subClick = container.querySelector(`[data-test="${subElementId}"]`)
      act(() => {
        if (subClick) fireEvent.click(subClick)
      })
      jest.runAllTimers()
      expect(mockFn).toHaveBeenCalledTimes(1)
      expect(mockFn).toHaveBeenLastCalledWith(result)
    },
  )
})
