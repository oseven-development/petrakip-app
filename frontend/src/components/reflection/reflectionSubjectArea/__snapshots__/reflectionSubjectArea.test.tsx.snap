// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main suite render sublist when click on 1 Fachinhalte 1`] = `
<div>
  <ion-list>
    <ion-item
      button="true"
      color="tertiary"
      data-id="1-fachinhalte"
      data-test="1-fachinhalte"
    >
      <ion-label
        data-id="1-fachinhalte"
      >
        1 Fachinhalte
      </ion-label>
      <ion-badge
        color="primary"
        data-id="1-fachinhalte"
      >
        1/4
      </ion-badge>
    </ion-item>
    <ion-item>
      <ion-text
        class="ion-padding-top ion-padding-bottom"
        color="medium"
      >
        Die Lehrperson verfügt über fachwissenschaftliches und fachdidaktisches Wissen, versteht die Inhalte, Strukturen und zentralen Forschungsmethoden ihrer Fachbereiche und sie kann Lernsituationen schaffen, die die fachwissenschaftlichen und fachdidaktischen Aspekte für die Lernenden bedeutsam machen.
      </ion-text>
    </ion-item>
    <ion-list>
      <ion-item
        button="true"
        data-id="1.1 Fachwissen und Didaktik"
        data-test="1.1-fachwissen-und-didaktik"
      >
        <ion-label
          data-id="1.1 Fachwissen und Didaktik"
        >
          1.1 Fachwissen und Didaktik
        </ion-label>
        <ion-icon
          color="medium"
          data-id="1.1 Fachwissen und Didaktik"
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Checkmark Circle</title><path d='M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z' stroke-miterlimit='10' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M352 176L217.6 336 160 272' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
          slot="end"
        />
      </ion-item>
      <ion-item
        button="true"
        data-id="1.2 Forschungsmethoden"
        data-test="1.2-forschungsmethoden"
      >
        <ion-label
          data-id="1.2 Forschungsmethoden"
        >
          1.2 Forschungsmethoden
        </ion-label>
        <ion-icon
          color="medium"
          data-id="1.2 Forschungsmethoden"
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Checkmark Circle</title><path d='M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z' stroke-miterlimit='10' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M352 176L217.6 336 160 272' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
          slot="end"
        />
      </ion-item>
      <ion-item
        button="true"
        data-id="1.3 Zusammenhänge erkennen"
        data-test="1.3-zusammenhänge-erkennen"
      >
        <ion-label
          data-id="1.3 Zusammenhänge erkennen"
        >
          1.3 Zusammenhänge erkennen
        </ion-label>
        <ion-icon
          color="medium"
          data-id="1.3 Zusammenhänge erkennen"
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Checkmark Circle</title><path d='M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z' stroke-miterlimit='10' class='ionicon-fill-none ionicon-stroke-width'/><path stroke-linecap='round' stroke-linejoin='round' d='M352 176L217.6 336 160 272' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
          slot="end"
        />
      </ion-item>
      <ion-item
        button="true"
        data-id="1.4 Auswahl von Zielen und Inhalten"
        data-test="1.4-auswahl-von-zielen-und-inhalten"
      >
        <ion-label
          data-id="1.4 Auswahl von Zielen und Inhalten"
        >
          1.4 Auswahl von Zielen und Inhalten
        </ion-label>
        <ion-icon
          color="success"
          data-id="1.4 Auswahl von Zielen und Inhalten"
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Checkmark Circle</title><path d='M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm108.25 138.29l-134.4 160a16 16 0 01-12 5.71h-.27a16 16 0 01-11.89-5.3l-57.6-64a16 16 0 1123.78-21.4l45.29 50.32 122.59-145.91a16 16 0 0124.5 20.58z'/></svg>"
          slot="end"
        />
      </ion-item>
    </ion-list>
    <ion-item
      button="true"
      data-id="2-entwicklungsprozesse"
      data-test="2-entwicklungsprozesse"
    >
      <ion-label
        color="medium"
        data-id="2-entwicklungsprozesse"
      >
        2 Entwicklungsprozesse
      </ion-label>
      <ion-badge
        color="primary"
        data-id="2-entwicklungsprozesse"
      >
        3/5
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="3-heterogenität-beim-lernen"
      data-test="3-heterogenität-beim-lernen"
    >
      <ion-label
        color="medium"
        data-id="3-heterogenität-beim-lernen"
      >
        3 Heterogenität beim Lernen
      </ion-label>
      <ion-badge
        color="primary"
        data-id="3-heterogenität-beim-lernen"
      >
        2/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="4-unterrichtsstrategien"
      data-test="4-unterrichtsstrategien"
    >
      <ion-label
        color="medium"
        data-id="4-unterrichtsstrategien"
      >
        4 Unterrichtsstrategien
      </ion-label>
      <ion-badge
        color="success"
        data-id="4-unterrichtsstrategien"
      >
        4/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="5-motivieren-und-anleiten"
      data-test="5-motivieren-und-anleiten"
    >
      <ion-label
        color="medium"
        data-id="5-motivieren-und-anleiten"
      >
        5 Motivieren und Anleiten
      </ion-label>
      <ion-badge
        color="primary"
        data-id="5-motivieren-und-anleiten"
      >
        3/5
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="6-kommunikation-und-moderation"
      data-test="6-kommunikation-und-moderation"
    >
      <ion-label
        color="medium"
        data-id="6-kommunikation-und-moderation"
      >
        6 Kommunikation und Moderation
      </ion-label>
      <ion-badge
        color="primary"
        data-id="6-kommunikation-und-moderation"
      >
        2/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="7-planen-und-evaluieren"
      data-test="7-planen-und-evaluieren"
    >
      <ion-label
        color="medium"
        data-id="7-planen-und-evaluieren"
      >
        7 Planen und Evaluieren
      </ion-label>
      <ion-badge
        color="primary"
        data-id="7-planen-und-evaluieren"
      >
        1/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="8-beobachtung-und-förderung"
      data-test="8-beobachtung-und-förderung"
    >
      <ion-label
        color="medium"
        data-id="8-beobachtung-und-förderung"
      >
        8 Beobachtung und Förderung
      </ion-label>
      <ion-badge
        color="primary"
        data-id="8-beobachtung-und-förderung"
      >
        3/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="9-reflexion"
      data-test="9-reflexion"
    >
      <ion-label
        color="medium"
        data-id="9-reflexion"
      >
        9 Reflexion
      </ion-label>
      <ion-badge
        color="primary"
        data-id="9-reflexion"
      >
        1/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="10-kooperation-umfeld"
      data-test="10-kooperation-umfeld"
    >
      <ion-label
        color="medium"
        data-id="10-kooperation-umfeld"
      >
        10 Kooperation Umfeld
      </ion-label>
      <ion-badge
        color="primary"
        data-id="10-kooperation-umfeld"
      >
        2/4
      </ion-badge>
    </ion-item>
  </ion-list>
  <ion-button
    data-id="Später wählen"
    data-test="choseLater"
  >
    Später wählen
  </ion-button>
</div>
`;

exports[`main suite render sublist when click on 4 Unterrichtsstrategien 1`] = `
<div>
  <ion-list>
    <ion-item
      button="true"
      data-id="1-fachinhalte"
      data-test="1-fachinhalte"
    >
      <ion-label
        color="medium"
        data-id="1-fachinhalte"
      >
        1 Fachinhalte
      </ion-label>
      <ion-badge
        color="primary"
        data-id="1-fachinhalte"
      >
        1/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="2-entwicklungsprozesse"
      data-test="2-entwicklungsprozesse"
    >
      <ion-label
        color="medium"
        data-id="2-entwicklungsprozesse"
      >
        2 Entwicklungsprozesse
      </ion-label>
      <ion-badge
        color="primary"
        data-id="2-entwicklungsprozesse"
      >
        3/5
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="3-heterogenität-beim-lernen"
      data-test="3-heterogenität-beim-lernen"
    >
      <ion-label
        color="medium"
        data-id="3-heterogenität-beim-lernen"
      >
        3 Heterogenität beim Lernen
      </ion-label>
      <ion-badge
        color="primary"
        data-id="3-heterogenität-beim-lernen"
      >
        2/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      color="tertiary"
      data-id="4-unterrichtsstrategien"
      data-test="4-unterrichtsstrategien"
    >
      <ion-label
        data-id="4-unterrichtsstrategien"
      >
        4 Unterrichtsstrategien
      </ion-label>
      <ion-badge
        color="success"
        data-id="4-unterrichtsstrategien"
      >
        4/4
      </ion-badge>
    </ion-item>
    <ion-item>
      <ion-text
        class="ion-padding-top ion-padding-bottom"
        color="medium"
      >
        Die Lehrperson versteht und verwendet eine Vielfalt von Unterrichtsstrategien, um bei den Lernenden stufengerechte, instrumentelle Fähigkeiten zu entwickeln und zu fördern.
      </ion-text>
    </ion-item>
    <ion-list>
      <ion-item
        button="true"
        data-id="4.1 Aufträge planen und formulieren"
        data-test="4.1-aufträge-planen-und-formulieren"
      >
        <ion-label
          data-id="4.1 Aufträge planen und formulieren"
        >
          4.1 Aufträge planen und formulieren
        </ion-label>
        <ion-icon
          color="success"
          data-id="4.1 Aufträge planen und formulieren"
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Checkmark Circle</title><path d='M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm108.25 138.29l-134.4 160a16 16 0 01-12 5.71h-.27a16 16 0 01-11.89-5.3l-57.6-64a16 16 0 1123.78-21.4l45.29 50.32 122.59-145.91a16 16 0 0124.5 20.58z'/></svg>"
          slot="end"
        />
      </ion-item>
      <ion-item
        button="true"
        data-id="4.2 Methodenvielfalt"
        data-test="4.2-methodenvielfalt"
      >
        <ion-label
          data-id="4.2 Methodenvielfalt"
        >
          4.2 Methodenvielfalt
        </ion-label>
        <ion-icon
          color="success"
          data-id="4.2 Methodenvielfalt"
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Checkmark Circle</title><path d='M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm108.25 138.29l-134.4 160a16 16 0 01-12 5.71h-.27a16 16 0 01-11.89-5.3l-57.6-64a16 16 0 1123.78-21.4l45.29 50.32 122.59-145.91a16 16 0 0124.5 20.58z'/></svg>"
          slot="end"
        />
      </ion-item>
      <ion-item
        button="true"
        data-id="4.3 Einsatz von Lernmaterialien"
        data-test="4.3-einsatz-von-lernmaterialien"
      >
        <ion-label
          data-id="4.3 Einsatz von Lernmaterialien"
        >
          4.3 Einsatz von Lernmaterialien
        </ion-label>
        <ion-icon
          color="success"
          data-id="4.3 Einsatz von Lernmaterialien"
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Checkmark Circle</title><path d='M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm108.25 138.29l-134.4 160a16 16 0 01-12 5.71h-.27a16 16 0 01-11.89-5.3l-57.6-64a16 16 0 1123.78-21.4l45.29 50.32 122.59-145.91a16 16 0 0124.5 20.58z'/></svg>"
          slot="end"
        />
      </ion-item>
      <ion-item
        button="true"
        data-id="4.4 Instrumentelle Fähigkeiten"
        data-test="4.4-instrumentelle-fähigkeiten"
      >
        <ion-label
          data-id="4.4 Instrumentelle Fähigkeiten"
        >
          4.4 Instrumentelle Fähigkeiten
        </ion-label>
        <ion-icon
          color="success"
          data-id="4.4 Instrumentelle Fähigkeiten"
          icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Checkmark Circle</title><path d='M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm108.25 138.29l-134.4 160a16 16 0 01-12 5.71h-.27a16 16 0 01-11.89-5.3l-57.6-64a16 16 0 1123.78-21.4l45.29 50.32 122.59-145.91a16 16 0 0124.5 20.58z'/></svg>"
          slot="end"
        />
      </ion-item>
    </ion-list>
    <ion-item
      button="true"
      data-id="5-motivieren-und-anleiten"
      data-test="5-motivieren-und-anleiten"
    >
      <ion-label
        color="medium"
        data-id="5-motivieren-und-anleiten"
      >
        5 Motivieren und Anleiten
      </ion-label>
      <ion-badge
        color="primary"
        data-id="5-motivieren-und-anleiten"
      >
        3/5
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="6-kommunikation-und-moderation"
      data-test="6-kommunikation-und-moderation"
    >
      <ion-label
        color="medium"
        data-id="6-kommunikation-und-moderation"
      >
        6 Kommunikation und Moderation
      </ion-label>
      <ion-badge
        color="primary"
        data-id="6-kommunikation-und-moderation"
      >
        2/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="7-planen-und-evaluieren"
      data-test="7-planen-und-evaluieren"
    >
      <ion-label
        color="medium"
        data-id="7-planen-und-evaluieren"
      >
        7 Planen und Evaluieren
      </ion-label>
      <ion-badge
        color="primary"
        data-id="7-planen-und-evaluieren"
      >
        1/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="8-beobachtung-und-förderung"
      data-test="8-beobachtung-und-förderung"
    >
      <ion-label
        color="medium"
        data-id="8-beobachtung-und-förderung"
      >
        8 Beobachtung und Förderung
      </ion-label>
      <ion-badge
        color="primary"
        data-id="8-beobachtung-und-förderung"
      >
        3/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="9-reflexion"
      data-test="9-reflexion"
    >
      <ion-label
        color="medium"
        data-id="9-reflexion"
      >
        9 Reflexion
      </ion-label>
      <ion-badge
        color="primary"
        data-id="9-reflexion"
      >
        1/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="10-kooperation-umfeld"
      data-test="10-kooperation-umfeld"
    >
      <ion-label
        color="medium"
        data-id="10-kooperation-umfeld"
      >
        10 Kooperation Umfeld
      </ion-label>
      <ion-badge
        color="primary"
        data-id="10-kooperation-umfeld"
      >
        2/4
      </ion-badge>
    </ion-item>
  </ion-list>
  <ion-button
    data-id="Später wählen"
    data-test="choseLater"
  >
    Später wählen
  </ion-button>
</div>
`;

exports[`main suite renders without crashing 1`] = `
<div>
  <ion-list>
    <ion-item
      button="true"
      data-id="1-fachinhalte"
      data-test="1-fachinhalte"
    >
      <ion-label
        data-id="1-fachinhalte"
      >
        1 Fachinhalte
      </ion-label>
      <ion-badge
        color="primary"
        data-id="1-fachinhalte"
      >
        1/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="2-entwicklungsprozesse"
      data-test="2-entwicklungsprozesse"
    >
      <ion-label
        data-id="2-entwicklungsprozesse"
      >
        2 Entwicklungsprozesse
      </ion-label>
      <ion-badge
        color="primary"
        data-id="2-entwicklungsprozesse"
      >
        3/5
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="3-heterogenität-beim-lernen"
      data-test="3-heterogenität-beim-lernen"
    >
      <ion-label
        data-id="3-heterogenität-beim-lernen"
      >
        3 Heterogenität beim Lernen
      </ion-label>
      <ion-badge
        color="primary"
        data-id="3-heterogenität-beim-lernen"
      >
        2/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="4-unterrichtsstrategien"
      data-test="4-unterrichtsstrategien"
    >
      <ion-label
        data-id="4-unterrichtsstrategien"
      >
        4 Unterrichtsstrategien
      </ion-label>
      <ion-badge
        color="success"
        data-id="4-unterrichtsstrategien"
      >
        4/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="5-motivieren-und-anleiten"
      data-test="5-motivieren-und-anleiten"
    >
      <ion-label
        data-id="5-motivieren-und-anleiten"
      >
        5 Motivieren und Anleiten
      </ion-label>
      <ion-badge
        color="primary"
        data-id="5-motivieren-und-anleiten"
      >
        3/5
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="6-kommunikation-und-moderation"
      data-test="6-kommunikation-und-moderation"
    >
      <ion-label
        data-id="6-kommunikation-und-moderation"
      >
        6 Kommunikation und Moderation
      </ion-label>
      <ion-badge
        color="primary"
        data-id="6-kommunikation-und-moderation"
      >
        2/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="7-planen-und-evaluieren"
      data-test="7-planen-und-evaluieren"
    >
      <ion-label
        data-id="7-planen-und-evaluieren"
      >
        7 Planen und Evaluieren
      </ion-label>
      <ion-badge
        color="primary"
        data-id="7-planen-und-evaluieren"
      >
        1/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="8-beobachtung-und-förderung"
      data-test="8-beobachtung-und-förderung"
    >
      <ion-label
        data-id="8-beobachtung-und-förderung"
      >
        8 Beobachtung und Förderung
      </ion-label>
      <ion-badge
        color="primary"
        data-id="8-beobachtung-und-förderung"
      >
        3/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="9-reflexion"
      data-test="9-reflexion"
    >
      <ion-label
        data-id="9-reflexion"
      >
        9 Reflexion
      </ion-label>
      <ion-badge
        color="primary"
        data-id="9-reflexion"
      >
        1/4
      </ion-badge>
    </ion-item>
    <ion-item
      button="true"
      data-id="10-kooperation-umfeld"
      data-test="10-kooperation-umfeld"
    >
      <ion-label
        data-id="10-kooperation-umfeld"
      >
        10 Kooperation Umfeld
      </ion-label>
      <ion-badge
        color="primary"
        data-id="10-kooperation-umfeld"
      >
        2/4
      </ion-badge>
    </ion-item>
  </ion-list>
  <ion-button
    data-id="Später wählen"
    data-test="choseLater"
  >
    Später wählen
  </ion-button>
</div>
`;
